### PR TITLE
Adjust memory page size for Arm architecture

### PR DESCRIPTION
**Pull request by Aakanksha0311**
In the context of Arm architecture, memory pages are different compared to x86. By default, x86 uses 2M memory pages, while Arm requires 512M memory pages.
This commit addresses an error related to attaching a device by adjusting the memory page size to 512M for Arm architecture. This change ensures compatibility and resolves the reported issue.

Test result for arm arch-
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : a74efc7fc3f3cd31c50b1e28743a2bf6366ff4a3
JOB LOG    : /var/lib/avocado/job-results/job-2023-09-11T03.22-a74efc7/job.log
 (1/1) type_specific.io-github-autotest-libvirt.memory_update_device.virtio_mem: PASS (72.93 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-09-11T03.22-a74efc7/results.html

Test result for x86:
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : 85eea8d991afd3c0fc1aca7b3102a7478bd6870e
JOB LOG    : /var/lib/avocado/job-results/job-2023-09-12T06.44-85eea8d/job.log
 (1/1) type_specific.io-github-autotest-libvirt.memory_update_device.virtio_mem: ERROR: VM 'avocado-vt-vm1' failed to start: error: Failed to start domain 'avocado-vt-vm1'\nerror: internal error: Unable to find any usable hugetlbfs mount for 524288 KiB(exit status: 1) (53.27 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-09-12T06.44-85eea8d/results.html
JOB TIME   : 53.73 s

### CODE REVIEW

**Comment by Yingshun** [comment]
Please upload test results for both x86_64 and arm arch. 
There are the same problems for memory_attach_device.virtio_mem, please fix up hem too. Thanks!

**Review by Yingshun — COMMENTED** [review]
Can you check with the feature owner if we can remove "set_num_huge_pages" in the memory_update_device test? It seems that its setup steps are similar to those of memory_attach_device.

BTW, plz 'Squash Commits'. This is to fix a problem about  block size setting, so please merge these 2 commits to one, thanks!

**Comment by Yingshun** [comment]
Please also upload test results for memory_attach_device on x86_64 and arm arch. Thx！

File: libvirt/tests/cfg/memory/memory_update_device.cfg [hunk 1]
```diff
@@ -13,7 +13,7 @@
             vm_attrs = {'max_mem_rt': 10485760, 'max_mem_rt_slots': 16, 'max_mem_rt_unit': 'KiB', 'vcpu': 4, 'cpu': {'numa_cell': [{'id': '0', 'cpus': '0-1', 'memory': '1048576', 'unit': 'KiB'}]}}
             aarch64:
                 vm_attrs = {'max_mem_rt': 10485760, 'max_mem_rt_slots': 16, 'max_mem_rt_unit': 'KiB', 'vcpu': 4, 'cpu': {'mode': 'host-passthrough', 'numa_cell': [{'id': '0', 'cpus': '0-3', 'memory': '524288', 'unit': 'KiB'}]}}
-            mem_device_attrs = {'mem_model': 'virtio-mem', 'target': {'requested_unit': 'KiB', 'size': 262144, 'node': 0, 'size_unit': 'KiB', 'requested_size': 131072, 'block_unit': 'KiB', 'block_size': 2048}, 'source': {'pagesize_unit': 'KiB', 'pagesize': 2048, 'nodemask': '0'}}
-            requested_size = 160MiB
+            mem_device_attrs = {'mem_model': 'virtio-mem', 'target': {'requested_unit': 'KiB', 'size': 1048576, 'node': 0, 'size_unit': 'KiB', 'requested_size': 524288, 'block_unit': 'KiB', 'block_size': 524288}, 'source': {'pagesize_unit': 'KiB', 'pagesize': 524288, 'nodemask': '0'}}
```
[the remark this answers is not in the document]
Comment (Aakanksha0311) [reply libvirt/tests/cfg/memory/memory_update_device.cfg:16]: Yes, It doesn't it fails on x86_64arch!

[the remark this answers is not in the document]
Comment (Yingshun) [reply libvirt/tests/cfg/memory/memory_update_device.cfg:16]: In that case, you can only update it for arm, plz refer to https://github.com/autotest/tp-libvirt/blob/master/libvirt/tests/cfg/memory/memory_update_device.cfg#L13~L15

**Comment by Aakanksha0311** [comment]
There were some complex errors while pushing new changes hence created a new MR and closing this one.

**Pull request closed without merging**